hphp/hack/test/typecheck/xhp_modifier/xhp_xhp_name.php

```diff
@@ -8,16 +8,10 @@
 
 xhp class foo {}
 
-xhp class foo:bar {}
```

Comment:
I'd like to have another PR up resolving this before we take this in: we're not going to be able to start migration. We're not going to be able to start any migration until `xhp class foo:bar` is valid

What class does the typechecker consider to be defined here?
What class does hh_single_compile emit?

Reply:
Wait, so we're going to support all of:

```
xhp class foo\bar {}
xhp class foo:bar {}
```

What's the benefit of changing `class :foo:bar` to `xhp class foo:bar`? Presumably there's an incremental migration flow you have in mind?

Reply:
`:` needs to be explicitly a namespace separator, not a sigil as well

at least one out of `xhp class foo:bar` and `xhp class foo\bar` is wanted to make it so we don't need to namespace everything

Reply:
```
namespace MyNS {

class :foo:bar {}

$_  = <:foo:bar />; // this doesn't refer to the class above

}
```

Reply:
`xhp class foo\bar {}` is not supported  - this need to be `xhp class foo:bar`